tst/com/amazon/corretto/crypto/provider/test/EvpKeyFactoryTest.java

```diff
@@ -740,6 +719,25 @@ private static class Samples<T> {
     }
   }
 
+  // This method is used to determine whether tests should use an alternate provider for a given
+  // algorithm. In cases where JCE doesn't support the requested algorithm, the alternate provider
+  // will be returned. In cases where JCE does support the requested algorithm, null will be
+  // returned.
+  private static Provider getAlternateProvider(String algorithm) {
+    // JCE doesn't support ML-DSA until JDK24, and BouncyCastle currently serializes ML-DSA private
```

Comment:
NP: Based on this, should the first condition below be `(algorithm.startsWith("ML-DSA") &&TestUtil.getJavaVersion() < 24)` analogous to the "Eds"?

Reply:
hm, yeah, that's a good idea